core/store.go

```diff
@@ -183,6 +183,15 @@ func Keys(p string) ([]string, error) {
 	return keys, err
 }
 
+// GetDbSize returns all the keys present in the db
+func GetDBSize() uint64 {
+	var noOfKeys uint64
+	withLocks(func() {
+		noOfKeys = uint64(len(keypool))
+	}, WithStoreRLock(), WithKeypoolLock())
```

Comment:
No need to take a lock on the store since we don't access it. Also, use `WithKeypoolRLock()` (read-lock for keypool instead of write lock)

Reply:
Review comment addressed...